File: frontend/classes/lon.php
<?php

class Lon extends Cordinate {

	function __construct($deg, $min, $sec) {
		
		if(!(-180 <= $deg && $deg <= 180)) {
			throw new Exception("Invalid longitude degree value $deg given, should be -180 <= deg <= 180");
		}
		
		if(!(0 <= $min && $min < 60)) {
			throw new Exception("Invalid longitude minute value $min given, should be 0 <= min <= 59");
		}
		
		if(!(0 <= $sec && $sec <= 60)) {
			throw new Exception("Invalid longitude second value $sec given, should be 0 <= sec <= 60");
		}
		
		$this->dec = $this->DMStoDEC($deg, $min, $sec);
		$this->rad = deg2rad($this->dec);
		
		parent::__construct($deg, $min, $sec);
	}
}

?>

File: radio/src/radio/Lon.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package radio;

import java.security.InvalidParameterException;

/**
 *
 * @author egidijus
 */
public class Lon extends Cordinate {
    public Lon(int deg, int min, int sec) {
        this.dec = super.DMStoDEC(deg, min, sec);
        this.rad = Math.toRadians(this.dec);
        super.Cordinate(deg, min, sec);
    }
    
    public Lon(double dec) {
        if(!(-90 <= dec && dec <= 90)) {
            throw new InvalidParameterException("Invalid latitude given, should be -90 <= deg <= 90");
	}
	
	this.dec = dec;
	this.rad = Math.toRadians(dec);

        short[] c = DECtoDMS(dec);
	super.Cordinate(c[0], c[1], c[2]);
    }
}
